src/components/network-map-tab.js

```diff
@@ -194,78 +183,123 @@ export const NetworkMapTab = ({
         []
     );
 
+    const getMissingEquipmentsPositions = useCallback(
+        (foundEquipmentPositions, allEquipments, fetchEquipmentCB) => {
+            let notFoundEquipmentsIds = [];
+            const foundEquipmentsIds = Array.from(
+                foundEquipmentPositions.keys()
+            );
+            allEquipments.forEach((s) => {
+                if (!foundEquipmentsIds.includes(s.id)) {
+                    notFoundEquipmentsIds.push(s.id);
+                }
+            });
+            if (notFoundEquipmentsIds.length === 0) {
+                return Promise.resolve([]);
+            }
+            setWaitingLoadGeoData(true);
+            return fetchEquipmentCB(
+                studyUuid,
+                currentNode?.id,
+                notFoundEquipmentsIds
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
     const reloadMapGeoData = useCallback(() => {
-        console.info(`Loading geo data of study '${studyUuid}'...`);
-        const substationPositions = fetchSubstationPositions(
-            studyUuid,
-            currentNode?.id
-        );
-        const linePositions = lineFullPath
-            ? fetchLinePositions(studyUuid, currentNode?.id)
-            : [];
-        setWaitingLoadGeoData(true);
-
-        Promise.all([substationPositions, linePositions])
-            .then((values) => {
-                const newGeoData = new GeoData();
-                newGeoData.setSubstationPositions(values[0]);
-                newGeoData.setLinePositions(values[1]);
-                setGeoData(newGeoData);
-                setWaitingLoadGeoData(false);
-            })
-            .catch(function (error) {
-                console.error(error.message);
-                setWaitingLoadGeoData(false);
-                setErrorMessage(
-                    intlRef.current.formatMessage(
-                        { id: 'geoDataLoadingFail' },
-                        { studyUuid: studyUuid }
-                    )
+        if (studyUuid && currentNode) {
+            if (
+                geoData &&
+                (geoData.substationPositionsById.size > 0 ||
+                    geoData.linePositionsById.size > 0)
+            ) {
+                const missingSubstationPositions =
```

Comment:
Need to do `setWaitingLoadGeoData(true);` before fetching missing Geo data I think.

Reply:
I wanted to do setWaitingLoadGeoData(true) only if we have some missing geodata. So I do it in getMissingEquipmentsPositions().
Maybe it's possible to remove setWaitingLoadGeoData(true)/setWaitingLoadGeoData(false) when we only fetch a delta of missing geodata as it is very quick ???